app/components/CustomForm.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import {
+  Form as RemixForm,
+  FormProps as RemixFormProps,
+} from "@remix-run/react";
+import { scrollToError } from "~/utils/scroll-to-error";
+
+type HTMLFormMethod = "post" | "get" | undefined;
```

Comment:
I dont think you need those custom types. You can just use the RemixFormProps instead. I did some testing and some changes(except component name) and came to this:
```ts
import React from "react";
import {
  Form as RemixForm,
  type FormProps as RemixFormProps,
} from "@remix-run/react";
import { scrollToError } from "~/utils/scroll-to-error";

export const CustomForm = React.forwardRef<HTMLFormElement, RemixFormProps>(
  (props, ref) => (
    <RemixForm
      ref={ref}
      {...props}
      onSubmit={(v) => {
        scrollToError();
        props?.onSubmit?.(v);
      }}
    />
  )
);

CustomForm.displayName = "CustomForm";
```